docs/content/en/docs/documentation/access-resources.md

```diff
@@ -0,0 +1,198 @@
+resources.
+
+However, once the event source has triggered a primary resource reconciliation, the associated reconciler needs to
+access the secondary resources which changes caused the reconciliation. Indeed, the information from the secondary
+resources might be needed during the reconciliation. For that purpose,  
+`InformerEventSource` maintains a reverse
+index [PrimaryToSecondaryIndex](https://github.com/operator-framework/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/DefaultPrimaryToSecondaryIndex.java),
+based on the result of the `SecondaryToPrimaryMapper`result.
+
+## Unified API for Related Resources
+
+To access all related resources for a primary resource, the framework provides an API to access the related
+secondary resources using the `Set<R> getSecondaryResources(Class<R> expectedType)` method of the `Context` object
+provided as part of the `reconcile` method.
+
+For `InformerEventSource`, this will leverage the associated `PrimaryToSecondaryIndex`. Resources are then retrieved
+from the informer's cache. Note that since all those steps work
+on top of indexes, those operations are very fast, usually O(1).
+
+While we've focused mostly on `InformerEventSource`, this concept can be extended to all `EventSources`, since
+[`EventSource`](https://github.com/operator-framework/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/EventSource.java#L93)
+actually implements the `Set<R> getSecondaryResources(P primary)` method that can be called from the `Context`.
+
+As there can be multiple event sources for the same resource types, things are a little more complex: the union of each
+event source results is returned.
+
+## Getting Resources Directly from Event Sources
+
+Note that nothing prevents you from directly accessing resources in the cache without going through
+`getSecondaryResources(...)`:
+
+```java
+public class WebPageReconciler implements Reconciler<WebPage> {
+
+    InformerEventSource<ConfigMap, WebPage> configMapEventSource;
+
+    @Override
+    public UpdateControl<WebPage> reconcile(WebPage webPage, Context<WebPage> context) {
+        // accessing resource directly from an event source 
+        var mySecondaryResource = configMapEventSource.get(new ResourceID("name", "namespace"));
+        // details omitted
+    }
+
+    @Override
+    public List<EventSource<?, WebPage>> prepareEventSources(EventSourceContext<WebPage> context) {
+        configMapEventSource = new InformerEventSource<>(
+                InformerEventSourceConfiguration.from(ConfigMap.class, WebPage.class)
+                        .withLabelSelector(SELECTOR)
+                        .build(),
+                context);
+
+        return List.of(configMapEventSource);
+    }
+}
+```
+
+## The Use Case for PrimaryToSecondaryMapper
```

Comment:
This whole section doesn't make much sense to me and needs to be clarified. The title mentions `PrimaryToSecondaryMapper` but barely talks about it.